### PR TITLE
feat: basic auth implemented

### PR DESCRIPTION
**Pull request by sakshi006**
Added basic authentication for the app.
When the user is not logged in, navbar will have 2 additional buttons, one for login and the other for signup.
Upon login, the navbar displays a single button for logginout

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -4,13 +4,17 @@ import Navbar from "./Components/Navbar/Navbar";
...
 
 function App() {
   return (
     <div className="App">
       <Navbar/>
       <Routes>
         <Route path="/" element={<Home/>}  />
+        <Route path="/login" element={<Login/>}  />
+        <Route path="/signup" element={<Signup/>}  />
         <Route path="mockman" element={<Mockman/>} />
```
Comment (Pratham82) [inline src/App.js:20]: 1. You can add a 404 page, if the page is not in the routes refer  this(https://ui.dev/react-router-handling-404-pages)

File: src/Pages/Authentication/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+  const addDummyUser = (e) => {
+    e.preventDefault();
+    setCredentials(dummyUser);
+  };
+  
+  const dataHandler = (e) => {
+    let name = e.target.name;
+    let value = e.target.value;
+    setCredentials({ ...credentials, [name]: value });
```
Comment (Pratham82) [inline src/Pages/Authentication/Login.jsx:24]: You can mdify the declaration of fields by this way.
`const {target:{ name, value } = e `

**Review by Pratham82 — APPROVED** [review]

**Comment by Pratham82** [comment]
Apart from these above changes looks good,

File: src/Pages/Authentication/Signup.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+const logInUser=async(e)=>{
+  e.preventDefault()
+  if(credentials.pass === credentials.cpass){
+    await axios.post('/api/auth/signup',{
+        firstName:'user',
+        lastName:'test',
+        email:credentials.email,
+        password:credentials.password
```
Comment (dev-enforced) [inline src/Pages/Authentication/Signup.jsx:29]: Can destructure credentials by passing it in the logInUser parameters as
```
const logInUser=async(e,{credentials:{email,password}})=>{
await axios.post("/api/auth/signup",{
firstName:"user",
lastName:"test",
email:email,
password:password
})
}
```
If you want you can change the names too of the parameters to be clear

File: src/Pages/Authentication/Signup.jsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  const navigate = useNavigate()
+  const [credentials,setCredentials]  = useState({
+      email:'',
+      pass:'',
+      cpass:''
+  })
+
+  const dataHandler=(e)=>{
+    let name = e.target.name
+    let value = e.target.value
```
Comment (dev-enforced) [inline src/Pages/Authentication/Signup.jsx:18]: Here also you can destructure as  or destructure it i
```
const dataHandler=({e:{target:{name,value}}})=>{
....
}
or you can do like this 
const dataHandler=(e)=>{
const {name,value}=e.target
}
```

File: src/App.js [hunk 5]
```diff
@@ -4,13 +4,17 @@ import Navbar from "./Components/Navbar/Navbar";
 import {Route,Routes} from "react-router-dom"
 import Home from "./Pages/Home/Home";
 import Mockman from "mockman-js";
+import { Login } from "./Pages/Authentication/Login";
+import Signup from "./Pages/Authentication/Signup";
```
Comment (dev-enforced) [inline src/App.js:8]: Can store all of these pages in a single file present in the pages folder of this repository that exports all these components from their location making it easy for you to import multiple components at the same time in one line

**Review by dev-enforced — APPROVED** [review]
The functionality is working as expected. Good work on the authentication @sakshi006 . Do work on the Suggestions given by @Pratham82. Gave some suggestions to the code base you have developed for this feature!!

File: src/App.js (hunk 5 above)
Comment (sakshi006) [reply src/App.js:8]: Hann, i will do that later while refactoring

**Pull request merged**